src/modules/flow/converter/converter.c

```diff
@@ -834,24 +837,179 @@ drange_to_string_convert(struct sol_flow_node *node, void *data, uint16_t port,
+        ptr++;
+    }
+
+    return true;
+}
+
+static bool
+validate_precision(const char *s)
+{
+    const char *ptr = s;
+    bool first = true;
+
+    while (*ptr) {
+        if (*ptr == '.') {
+            if (first)
+                first = false;
+            else
+                return false;
+        } else if (!isdigit((uint8_t)*ptr))
+            return false;
+        ptr++;
+    }
+
+    return true;
+}
+
+static bool
+validate_drange_conversion_specifier(const char *s)
+{
+    if ((s[0] == 'e' || s[0] == 'f' || s[0] == 'F' || s[0] == 'g' ||
+        s[0] == 'G' || s[0] == 'a' || s[0] == 'A') && s[1] == '\0')
+        return true;
+
+    return false;
+}
+
+static bool
+validate_irange_conversion_specifier(const char *s)
+{
+    if ((s[0] == 'd' || s[0] == 'i' || s[0] == 'o' || s[0] == 'u' ||
+        s[0] == 'x' || s[0] == 'X' || s[0] == 'c') && s[1] == '\0')
+        return true;
+
+    return false;
+}
+
 static int
 drange_to_string_open(struct sol_flow_node *node,
     void *data,
     const struct sol_flow_node_options *options)
 {
     struct string_converter *mdata = data;
     const struct sol_flow_node_type_converter_float_to_string_options *opts;
+    bool prefix = false, suffix = false, flags = false, f_width = false,
+        precision = false, valid_conv = false;
 
     SOL_FLOW_NODE_OPTIONS_SUB_API_CHECK(options,
         SOL_FLOW_NODE_TYPE_CONVERTER_FLOAT_TO_STRING_OPTIONS_API_VERSION,
         -EINVAL);
 
     opts = (const struct sol_flow_node_type_converter_float_to_string_options *)options;
 
-    mdata->format = strdup(opts->format_spec);
-    SOL_NULL_CHECK(mdata->format, -ENOMEM);
+    if (opts->format_prefix) {
+        if (!validate_affix(opts->format_prefix))
+            SOL_WRN("Invalid format prefix provided: %s,"
+                " proceeding with an empty one", opts->format_prefix);
+        else
+            prefix = true;
+    }
+    if (opts->format_suffix) {
+        if (!validate_affix(opts->format_suffix))
+            SOL_WRN("Invalid format suffix provided: %s,"
+                " proceeding with an empty one", opts->format_suffix);
+        else
+            suffix = true;
+    }
+    if (opts->format_flags) {
+        if (!validate_flags(opts->format_flags))
+            SOL_WRN("Invalid format flags provided: %s,"
+                " proceeding with no flags", opts->format_flags);
+        else
+            flags = true;
+    }
+    if (opts->format_field_width) {
+        if (!validate_field_width(opts->format_field_width))
+            SOL_WRN("Invalid format field width provided: %s,"
+                " proceeding with no field width", opts->format_field_width);
+        else
+            f_width = true;
+    }
+    if (opts->format_precision) {
+        if (!validate_precision(opts->format_precision))
+            SOL_WRN("Invalid format precision provided: %s,"
+                " proceeding with no precision", opts->format_precision);
+        else
+            precision = true;
+    }
```

Comment:
There's the negative field width case, but yeah, we're covered with integers. I'll change that, thanks.